src/components/layout/Layout/Layout.tsx

```diff
@@ -22,6 +31,11 @@ function Layout() {
       <div className="footer">
         <Footer />
       </div>
+      <ConnectWalletModal show={showConnectModal} />
+      {/* change to hook for display management */}
+      <CreateAntModal
+        show={showCreateAntModal && walletAddress !== undefined}
```

Comment:
i still think the `CreateAntModal` hook should check for `walletAddress` and dispatch the connect modal there rather than other places we are dispatching both.

Reply:
changed